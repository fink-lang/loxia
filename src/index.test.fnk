{generate} = import '.'
{fink2js} = import './testing'
{describe, it, to_throw} = import './testing/jest'


describe:: 'errors', fn:
  it:: 'throws with code snippet', fn:
    (
      fn: fink2js(`
        123 = foo`
      )
    ) to_throw `
      test.fnk:1:0
      1| 123 = foo
         ^

      Unable to transform 'assign ='.`

    (
      fn: generate({
        type: 'test',
        op: null,
        loc: {
          start: {pos: 0, line: 1, column: 0},
          end: {pos: 9, line: 1, column: 9}
        }
      }, 'test.fnk', 'foobar')

    ) to_throw `
      test.fnk:1:0
      1| foobar
         ^

      Unable to transform 'test'.

      Unknown expression`


