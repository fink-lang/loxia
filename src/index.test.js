import {generate} from '.';
import {unindent_text, fink2js} from './testing';


test('errors', ()=> {
  expect(
    ()=> fink2js(`
      123 = foo`
    )
  ).toThrow(unindent_text(`
    test.fnk:1:0
    1| 123 = foo
       ^

    Unable to transform 'assign ='.`)
  );


  expect(
    ()=> generate({
      type: 'test',
      op: null,
      loc: {
        start: {pos: 0, line: 1, column: 0},
        end: {pos: 9, line: 1, column: 9}
      }
    }, 'test.fnk', 'foobar')

  ).toThrow(unindent_text(`
    test.fnk:1:0
    1| foobar
       ^

    Unable to transform 'test'.

    Unknown expression`)
  );
});
