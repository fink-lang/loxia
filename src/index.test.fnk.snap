// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module types compiles to cjs 1`] = `
"require(\\"foobar\\");

const foo = require(\\"shrub\\");

const {
  bar: spam,
  ni
} = require(\\"ni\\");

const shrub = ni;
exports.shrub = shrub;
exports.default = shrub;"
`;

exports[`module types compiles to mjs 1`] = `
"import \\"foobar\\";
import foo from \\"shrub\\";
import { bar as spam, ni } from \\"ni\\";
export const shrub = ni;
export default shrub;"
`;
